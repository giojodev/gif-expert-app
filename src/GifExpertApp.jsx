import { useState } from "react"

const GifExpertApp = () => {
  
  const [categories, setCategories] = useState(['One Piece', 'Naruto','Bleach']);

  return (
    <>
        <h1>GifExpertApp </h1>


        <ol>
          {categories.map(e=>{
            return <li key={e}>{e}</li>
          })}
        
        
          
        </ol>
    </>
  )
}

export default GifExpertApp
