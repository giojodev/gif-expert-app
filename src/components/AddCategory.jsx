import { useState } from "react"

export const AddCategory = () =>{
    
    const [inputValue, setinputValue] = useState('One Piece')
    
    const onInputChange = (event)=> {
        setinputValue(event.target.value);
    }

    const onSubmit = (event) =>{
        event.preventDefault();

        console.log(inputValue);
    }

    return (
      <form onSubmit={(event)=>onSubmit(event)}>
        <input
          id="txtCategories"
          type="text"
          placeholder="Buscar GIF"
          value={inputValue}
          onChange={(event) => onInputChange(event)}
        ></input>
      </form>
    );
}