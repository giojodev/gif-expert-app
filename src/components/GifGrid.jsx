// import { Grid,GridCol } from "@mantine/core"
import {getGifs} from '../helpers/getGifs'
import { useEffect } from 'react';

export const GifGrid = ({category})=>{

  useEffect(() => {

    getGifs(category);
  }, []);

  return (
  <>
    {/* <Grid> */}
      {category.map((e) => (
        // <GridCol span={4}>
          <div key={e}>
            <h3>{e}</h3>
            <p>{e}</p>
          </div>
        // </GridCol>
      ))}
    {/* </Grid> */}
  </>
);
}