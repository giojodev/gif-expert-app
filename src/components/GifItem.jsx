import { Card,Image,Text,Group, Divider } from "@mantine/core" 

export const GifItem = ({title,url,id}) =>{

    console.log({title,url,id});

    return(
        <>
        {/* <div className="card">
            <img
                src={url}
                alt={title}
            />
            
            <h3>{title}</h3>
        </div> */}
            <Card  padding="lg" radius="md" withBorder >
                <Card.Section>
                    <Image 
                     src={url}
                     height={180}
                     alt={title}
                     padding={10}
                    />
                </Card.Section>
                <Text size="md" c="dark" >
                    {title}
                </Text>
            </Card>
        </>
    )
}