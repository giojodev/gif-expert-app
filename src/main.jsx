import React from 'react'
import ReactDOM from 'react-dom/client'
import GifExpertApp from './GifExpertApp'
import './styles.css'
import '@mantine/core/styles.css';
;

ReactDOM.createRoot(document.getElementById("root")).render(

    <React.StrictMode>
      <GifExpertApp />
    </React.StrictMode>
    

);
