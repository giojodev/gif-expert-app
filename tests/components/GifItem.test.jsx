import {render} from '@testing-library/react';
import { GifItem } from '../../src/components';


describe('Pruebas en GiftItem', () => {
    test('Debe de hacer match con el snapshot', () => {
        const titulo= 'Saitama';
        const url2 = 'https://onepunch.com/saitam.jpg';
        const { asFragment } = render(<GifItem title={titulo} url={url2} />);
      
          expect(asFragment).toMatchSnapshot();
    });
});