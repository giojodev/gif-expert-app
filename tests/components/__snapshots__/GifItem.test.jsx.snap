// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GiftItem Debe de hacer match con el snapshot 1`] = `[Function]`;
